webapp/components/wpt-runs.html

```diff
@@ -184,16 +185,26 @@
           }, {});
 
           // We flatten into an array of objects so Polymer can deal with them.
-          const firstRunTime = x => Object.values(x.runs)[0][0].created_at;
-          const flattened = Object.entries(testRunsBySha)
-            .map(({0: sha, 1: runs}) => ({ sha, runs }))
-            .sort((a, b) => firstRunTime(a) < firstRunTime(b) ? 1 : -1);
+          const firstRunTime = runs => {
+            return Array.from(Object.values(runs))
+              .reduce((oldest, runs) => {
+                for (const time of runs.map(r => new Date(r.time_start))) {
+                  if (time < oldest) {
+                    oldest = time;
+                  }
+                }
+                return oldest;
+              }, new Date()); // Existing runs should be historical...
+          };
+          const flattened = Array.from(Object.entries(testRunsBySha))
+            .map(({0: sha, 1: runs}) => ({ sha, runs, firstRunTime: firstRunTime(runs) }))
```

Comment:
Unless there's more than meets the eye here, `({0: sha, 1: runs})` can be `([sha,runs])`. (`Object.entries` returns array of arrays, which also means that `Array.from` isn't needed.)